backend/geonature/core/notifications/models.py

```diff
@@ -142,3 +143,26 @@ class NotificationRule(db.Model):
     method = relationship(NotificationMethod)
     category = relationship(NotificationCategory)
     user = db.relationship(User)
+
+    @qfilter(query=True)
+    def filter_by_role_with_defaults(cls, id_role=None, **kwargs):
```

Comment:
```suggestion
    def filter_by_role(cls, *, query, id_role=None):
```
Est-ce que le `_with_defaults` n’est pas un peu lourd ?